pages/accounts/profile.py

```diff
@@ -272,7 +267,108 @@ class InactiveOption(Region):
     class PopupConsole(Region):
         """Popup console interaction."""
 
-        _root_locator = (By.CSS_SELECTOR, '.modal-content')
+        _users_locator = (By.LINK_TEXT, 'Users')
+        _misc_locator = (By.LINK_TEXT, 'Misc')
+        _links_locator = (By.LINK_TEXT, 'Links')
+        _full_console_locator = (By.LINK_TEXT, 'Full Console >>')
+
+        @property
+        def misc(self):
+            """Goes to misc tab of the pop up console."""
+            self.find_element(*self._misc_locator).click()
+            return self.Misc(self)
+
+        @property
+        def users(self):
+            """Goes to user tab of the pop up console."""
+            self.find_element(*self._users_locator).click()
+            return self.Users(self)
+
+        @property
+        def links(self):
+            """Goes to links tab of the pop up console."""
+            self.find_element(*self._links_locator).click()
+            return self.Links(self)
+
+        def full_console(self):
+            """Goes to full_console tab of the pop up console."""
+            self.find_element(*self._full_console_locator).click()
+            return self
+
+        class Misc(Region):
+            """Misc section."""
+
+            _users_locator = (By.LINK_TEXT, 'Users')
+            _links_locator = (By.LINK_TEXT, 'Links')
+            _full_console_locator = (By.LINK_TEXT, 'Full Console >>')
+
+            def go_to_user_section(self):
+                """Go to user section on the tab."""
+                self.find_element(*self._users_locator).click()
+                return self
+
+            def go_to_links_section(self):
+                """Go to links section on the tab."""
+                self.find_element(*self._links_locator).click()
+                return self
+
+            def go_to_full_section(self):
+                """Go to full section on the tab."""
+                self.find_element(*self._full_console_locator).click()
+                return self
+
+        class Users(Region):
+            """User section."""
+
+            _search_bar_locator = (By.ID, 'search_terms')
+            _search_button_locator = (By.NAME, 'commit')
+
+            @property
+            def search_empty(self):
+                """Search given string."""
+                self.find_element(*self._search_bar_locator).send_keys(
+                    "0teacher")
+                self.find_element(*self._search_button_locator).click()
+                sleep(1)
+                return self.Results(self)
```

Comment:
Whoops...this looks test code in the object; the object should have a way to search for a passed parameter. In the test code you can search for a specific string ("teacher" is a good one because it will return results for all instance sets).
Additionally, the return should be a list of search results. Look to https://github.com/openstax/os-automation/blob/master/pages/utils/email.py#L70-L82 for an example of a repeated region.